drivers/SmartThings/sonos/src/api/sonos_connection.lua

```diff
@@ -155,11 +168,26 @@ end
 ---@param sonos_conn SonosConnection
 local function _spawn_reconnect_task(sonos_conn)
   log.debug("Spawning reconnect task for ", sonos_conn.device.label)
+  local token_receive_handle, err = sonos_conn.driver:get_oauth_token_receive_handle()
+  if not token_receive_handle then
+    log.warn(string.format("error creating oauth token receive handle for respawn task: %s", err))
+  end
   cosock.spawn(function()
     local backoff = backoff_builder(60, 1, 0.1)
     while not sonos_conn:is_running() do
-      local start_success = sonos_conn:start()
-      if start_success then return end
+      if sonos_conn.driver.waiting_for_token and token_receive_handle then
+        local token, channel_error = token_receive_handle:receive()
```

Comment:
I don't know that we actually want to handle this in a loop; that said, I *believe* that this won't *actually* loop.

`get_oauth_token()` calls through to `security.get_sonos_oauth_token()`, which we have designed to be completely out-of-band async. So we fire-and-forget, with the return value either being nothing or an error due to perms/API compat.

The `token_receive_handle` is half of a cosock channel. We haven't put a timeout on it, and it's on a non-main thread, so it should just yield forever until it gets something. And that "something" should only come through if the send half of the channel gets dropped, or if the token arrives on the environment info update in the augmented store.

Reply:
I'll have to think about this, but I don't know if _no_ timeout is truly what we want.  I could see a very large timeout, but at the same time I understand the potential load caused by thousands of hubs repeatedly checking for a token when the user doesn't link their account.